<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>H5FDdsmPanel</class>
 <widget class="QWidget" name="H5FDdsmPanel">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>526</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QCheckBox" name="lockSettings">
     <property name="text">
      <string>Lock/Unlock Settings</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="dsmTypeBox">
     <property name="title">
      <string>DSM type</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <widget class="QRadioButton" name="dsmIsServer">
        <property name="text">
         <string>Server</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="dsmIsClient">
        <property name="text">
         <string>Client</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="dsmIsStandalone">
        <property name="text">
         <string>Standalone</string>
        </property>
        <property name="checked">
         <bool>false</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="dsmSettingBox">
     <property name="title">
      <string>HDF5 DSM Settings</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <property name="horizontalSpacing">
       <number>6</number>
      </property>
      <item row="4" column="1" colspan="2">
       <widget class="QComboBox" name="dsmInterCommType">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="currentIndex">
         <number>0</number>
        </property>
        <item>
         <property name="text">
          <string>Sockets</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>MPI</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>MPI_RMA</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>DMAPP</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="2" column="1" colspan="2">
       <widget class="QSpinBox" name="dsmInterCommPort">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="maximum">
         <number>65535</number>
        </property>
        <property name="singleStep">
         <number>1000</number>
        </property>
        <property name="value">
         <number>22000</number>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="dsmInterCommTypeLabel">
        <property name="minimumSize">
         <size>
          <width>150</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>Inter-Communicator</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="dsmSizeLabel">
        <property name="minimumSize">
         <size>
          <width>150</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>Buffer Size/Node (in MB)</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1" colspan="2">
       <widget class="QSpinBox" name="dsmSize">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="maximum">
         <number>65535</number>
        </property>
        <property name="value">
         <number>128</number>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="dsmServerNameLabel">
        <property name="minimumSize">
         <size>
          <width>150</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>Host (name/IP address)</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QComboBox" name="dsmServerName">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="currentIndex">
         <number>0</number>
        </property>
        <item>
         <property name="text">
          <string>default</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>agno.staff.cscs.ch</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>dino.staff.cscs.ch</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>breno.staff.cscs.ch</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>horus.login.cscs.ch</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>localhost</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="1" column="2">
       <widget class="QToolButton" name="addDsmServer">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>...</string>
        </property>
        <property name="icon">
         <iconset>
          <normaloff>:/pqWidgets/Icons/pqConnect24.png</normaloff>:/pqWidgets/Icons/pqConnect24.png</iconset>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="dsmInterCommPortLabel">
        <property name="minimumSize">
         <size>
          <width>150</width>
          <height>0</height>
         </size>
        </property>
        <property name="text">
         <string>Port number</string>
        </property>
       </widget>
      </item>
      <item row="5" column="0" colspan="3">
       <widget class="QCheckBox" name="staticInterCommBox">
        <property name="text">
         <string>Use Static Inter-Communicator</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QLabel" name="dsmDistributionTypeLabel">
        <property name="text">
         <string>Distribution Type</string>
        </property>
       </widget>
      </item>
      <item row="6" column="1" colspan="2">
       <widget class="QComboBox" name="dsmDistributionType">
        <item>
         <property name="text">
          <string>Contiguous</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Block Cyclic</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Random Block Cyclic</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="7" column="0">
       <widget class="QLabel" name="dsmBlockLengthLabel">
        <property name="text">
         <string>Block Length</string>
        </property>
       </widget>
      </item>
      <item row="7" column="1" colspan="2">
       <widget class="QSpinBox" name="dsmBlockLength">
        <property name="minimum">
         <number>256</number>
        </property>
        <property name="maximum">
         <number>4194304</number>
        </property>
        <property name="singleStep">
         <number>256</number>
        </property>
        <property name="value">
         <number>2048</number>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="sourceBox">
     <property name="enabled">
      <bool>true</bool>
     </property>
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>Active Source Object</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="0" column="2" colspan="2">
       <widget class="QPushButton" name="writeToDSM">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="text">
         <string>Write Active Source to DSM</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0" colspan="2">
       <widget class="QLineEdit" name="infoTextOutput">
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
